lib/client/kube/kube.go

```diff
@@ -0,0 +1,79 @@
+// Teleport
+// Copyright (C) 2023  Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package kube
+
+import (
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/tlsca"
+)
+
+var log = logrus.WithFields(logrus.Fields{
+	trace.Component: teleport.ComponentKubeClient,
+})
+
+// CheckIfCertsAreAllowedToAccessCluster evaluates if the new cert created by the user
+// to access kubeCluster has at least one kubernetes_user or kubernetes_group
+// defined. If not, it returns an error.
+// This is a safety check in order to print a better message to the user even
+// before hitting Teleport Kubernetes Proxy.
+func CheckIfCertsAreAllowedToAccessCluster(k *client.Key, rootCluster, teleportCluster, kubeCluster string) error {
```

Comment:
I'm not sure if this is the best place for this function, so please advise if you have a better idea.

It also makes me squirm that when importing this package, I import it as `kubeclient`, even though it does not contain any kind of kube client at the moment. It's more like just a single util function. But that's because `lib/client` is so big, if it was more targeted then I'm sure there would be a relevant `kubeclient` to put this in.

So I'm not sure if I should name the packages according to current meaning or future intent here.

There's `lib/kube/utils` already, so I tried to put this function there, but I ended up with a cyclic import.

Maybe I should just import it as `clientkube` instead?